patches/gwconditionallogicdates.0001.fix.php-warning-access-array-offset-on-value-of-type-null.patch

```diff
@@ -0,0 +1,29 @@
+From 2b7dd0e0b141987c12f82fc63838542110b9b060 Mon Sep 17 00:00:00 2001
+From: Nabi <artsanaat@gmail.com>
+Date: Wed, 13 Dec 2023 12:29:36 +0330
+Subject: [PATCH] Fixed Trying to access array offset on value of type null in
+ gwconditionallogicdates.
+
+---
+ includes/class-gw-conditional-logic-date-fields.php | 5 ++++-
+ 1 file changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/includes/class-gw-conditional-logic-date-fields.php b/includes/class-gw-conditional-logic-date-fields.php
+index da871fa6e..0003877fe 100644
+--- a/includes/class-gw-conditional-logic-date-fields.php
++++ b/includes/class-gw-conditional-logic-date-fields.php
+@@ -586,7 +586,10 @@ if ( ! class_exists( 'GWConditionalLogicDateFields' ) ) {
+ 			 * @see https://secure.helpscout.net/conversation/964227500/13854?folderId=14965
+ 			 */
+ 			if ( ! self::is_valid_timestamp( $target_value ) ) {
+-				$temp_rules   = self::convert_conditional_logic_rules( array( $rule ), $source_field->formId );
++				// In this case ($rule['fieldId'] == '_gpcld_current_time') we don't have a $source_field object.
++				// So we pass the form id from $this->current_form that was set in set_current_form().
++				$formId = $rule['fieldId'] == '_gpcld_current_time' ? $this->current_form['id'] : $source_field['formId'];
++				$temp_rules   = self::convert_conditional_logic_rules( array( $rule ),  $formId);
```

Comment:
Seems like `'_gpcld_current_time'` a custom/project-specific field?  Can't we simply check whether `$source_field` is empty to achieve the same?
```suggestion
+				// If there is no $source_field object, pass the form ID from set_current_form().
+				$temp_rules   = self::convert_conditional_logic_rules( array( $rule ), ! empty( $source_field ) ? $source_field->formId : $this->current_form['id'] );
```